reactnative-todo/ios/reactnativetodo/FRAuthSampleStructs.swift

```diff
@@ -0,0 +1,250 @@
+  var failedPolicies: [RawFailedPolicies]?
+
+  /**
+   * Raw journey/tree JSON response from ForgeRock server
+   */
+  var response: String
+
+  init(callback: Callback) {
+    self.type = callback.type
+
+    if let thisCallback = callback as? SingleValueCallback {
+      self.prompt = thisCallback.prompt
+      self.inputNames = [thisCallback.inputName!]
+    }
+
+    if let thisCallback = callback as? KbaCreateCallback {
+      self.prompt = thisCallback.prompt
+      self.predefinedQuestions = thisCallback.predefinedQuestions
+      self.inputNames = thisCallback.inputNames
+    }
+
+    if let thisCallback = callback as? ChoiceCallback {
+      self.choices = thisCallback.choices
+      self.inputNames = [thisCallback.inputName!]
+    }
+
+    if let thisCallback = callback as? AbstractValidatedCallback {
+      if let policyDictionary = thisCallback.policies,
+        let policiesJSON = try? policyDictionary.toJson() {
+        
+        let jsonData = Data(policiesJSON.utf8)
+
+        do {
+          self.policies = try JSONDecoder().decode(RawPolicies.self, from: jsonData)
+        }
+        catch {
+          print(error)
+        }
+      }
+      if let failedPolicies = thisCallback.failedPolicies {
+        self.failedPolicies = []
+        for failedPolicy in failedPolicies {
+          var paramsDictionary = [String: FlexibleType]()
+          if let params = failedPolicy.params {
+            let newDictionary = params.mapValues { value -> FlexibleType in
+              if let str = value as? String {
+                return FlexibleType(str, originalType: .String)
+              } else if let str = value as? Int {
+                return FlexibleType(String(str), originalType: .Int)
+              } else if let str = value as? Double {
+                return FlexibleType(String(str), originalType: .Double)
+              } else if let str = value as? Bool {
+                return FlexibleType(String(str), originalType: .Bool)
+              } else {
+                return FlexibleType("", originalType: .String)
+              }
+            }
+            paramsDictionary = newDictionary
+          }
+          self.failedPolicies?.append(
+            RawFailedPolicies(
+              propertyName: self.prompt,
+              params: paramsDictionary,
+              policyRequirement: failedPolicy.policyRequirement,
+              failedDescription: failedPolicy.failedDescription()
+            )
+          )
+        }
+      }
+    }
+
+    if let jsonData = try? JSONSerialization.data(withJSONObject: callback.response, options: .prettyPrinted),
+      let jsonString = String(data: jsonData, encoding: .utf8) {
+      self.response = jsonString
+    } else {
+      self.response = ""
+    }
+  }
+}
+
+public struct Response: Codable {
+  var authId: String?
+  var callbacks: [RawCallback]?
+  var status: Int?
+}
+
+public struct RawCallback: Codable {
+  var type: String?
+  var input: [RawInput]?
+  var _id: Int?
```

Comment:
Usually in swift we never start variable names with underscore.

Reply:
I agree, but this is to parse the response as it is coming from AM. I could have the coding keys but this is minimal impact for the rest of the code. Also, this Codable is only used for a very specific reason for parsing responses in the bridge